libs/perun/components/src/lib/ban-on-entity-list/ban-on-entity-list.component.ts

```diff
@@ -0,0 +1,170 @@
+import { Component, EventEmitter, Input, Output, ViewChild } from '@angular/core';
+import {
+  customDataSourceFilterPredicate,
+  customDataSourceSort,
+  downloadData,
+  getDataForExport,
+  TABLE_ITEMS_COUNT_OPTIONS,
+  TableWrapperComponent,
+} from '@perun-web-apps/perun/utils';
+import {
+  EnrichedBanOnFacility,
+  EnrichedBanOnResource,
+  EnrichedBanOnVo,
+} from '@perun-web-apps/perun/openapi';
+import { SelectionModel } from '@angular/cdk/collections';
+import { GuiAuthResolver, TableCheckbox } from '@perun-web-apps/perun/services';
+import { MatTableDataSource } from '@angular/material/table';
+import { MatSort } from '@angular/material/sort';
+
+export type EnrichedBan = EnrichedBanOnVo | EnrichedBanOnFacility | EnrichedBanOnResource;
+export type BanOnEntityListColumn =
+  | 'select'
+  | 'banId'
+  | 'targetName'
+  | 'targetId'
+  | 'subjectName'
+  | 'subjectId'
+  | 'description'
+  | 'validity'
+  | 'edit';
+
+@Component({
+  selector: 'perun-web-apps-ban-on-entity-list',
+  templateUrl: './ban-on-entity-list.component.html',
+  styleUrls: ['./ban-on-entity-list.component.scss'],
+})
+export class BanOnEntityListComponent {
+  @Input() selection = new SelectionModel<EnrichedBan>(false, []);
+  @Input() tableId: string;
+  @Input() updatePolicy: string;
+  @Input() pageSizeOptions = TABLE_ITEMS_COUNT_OPTIONS;
+  @Input() columns: BanOnEntityListColumn[] = [
+    'select',
+    'banId',
+    'targetId',
+    'targetName',
+    'subjectId',
+    'subjectName',
+    'description',
+    'validity',
+    'edit',
+  ];
+  @Output() updateBan = new EventEmitter<EnrichedBan>();
+  @ViewChild(TableWrapperComponent, { static: true }) tableWrapper: TableWrapperComponent;
+  @ViewChild(MatSort, { static: true }) sort: MatSort;
+
+  dataSource: MatTableDataSource<EnrichedBan>;
+  target: string;
+  subject: string;
+
+  constructor(private tableCheckbox: TableCheckbox, private authResolver: GuiAuthResolver) {}
+
+  @Input() set bans(bans: EnrichedBan[]) {
+    if (!this.dataSource) {
+      this.dataSourceInit(bans);
+    }
+    if (bans.length !== 0) {
+      this.setHeaderLabels(bans[0]);
+    }
+
+    this.dataSource.data = bans;
+  }
+
+  @Input() set filter(value: string) {
+    this.dataSource.filter = value;
+  }
+
+  @Input() set displayedColumns(columns: BanOnEntityListColumn[]) {
+    if (!this.authResolver.isPerunAdminOrObserver()) {
+      columns = columns.filter((column) => !column.endsWith('Id'));
+    }
+    this.columns = columns;
+  }
+
+  getDataForColumn = (data: EnrichedBan, column: BanOnEntityListColumn): string => {
+    switch (column) {
+      case 'banId':
+        return String(data.ban.id);
+      case 'subjectName': {
+        if (this.isFacilityBan(data)) return data.facility.name;
+        else if (this.isResourceBan(data)) return data.resource.name;
+        else return data.vo.name;
+      }
+      default: {
+        return String(data[column]);
```

Comment:
This is why filtering/sorting of the list fails IMO, column names don't match the names of the attributes